### PR TITLE
fix a number of clang tidy warnings

### PR DESCRIPTION
**Pull request by benoit-pierre**
This change is [Reviewable](https://reviewable.io/reviews/koreader/crengine/596)

### CODE REVIEW

File: crengine/src/hyphman.cpp [hunk 1]
```diff
@@ -1478,8 +1478,8 @@ lString32 UserHyphDict::getHyphenation(const char *word)
 
     lString32 word_str(orig_word_str.c_str() + start, end - start);
     size_t len = word_str.length();
-    lUInt16 widths[len+2];
-    lUInt8 flags[len+2];
+    lUInt16 widths[len+2]; // NOLINT(clang-diagnostic-vla-cxx-extension)
```
Comment (Frenzie) [inline crengine/src/hyphman.cpp:1481]: Clang extension in this context means Clang *and* GCC extension?

Comment (benoit-pierre) [reply crengine/src/hyphman.cpp:1481]: Yes.

**Comment by poire-z** [comment]
https://github.com/koreader/crengine/pull/596/commits/a30ca84221c9ca1cbb6f87df6c43ddaccf8f9bfa and https://github.com/koreader/crengine/pull/596/commits/7c7cec218aec9caf5ced35e5f19a3d3894a97670 and https://github.com/koreader/crengine/pull/596/commits/e68a446a84334131b4e1172c9d1bc9025c7970f5 could probably be merged into the same commit.

**Review by poire-z — APPROVED** [review]

**Comment by Frenzie** [comment]
As an aside, do you suppose it could be of interest to add a `flawfinder` check?
https://dwheeler.com/flawfinder/


<details>
<summary>Sample output on current master</summary>

```
Flawfinder version 2.0.19, (C) 2001-2019 David A. Wheeler.
Number of rules (primarily dangerous function names) in C/C++ ruleset: 222
Examining lvstring.cpp

FINAL RESULTS:

lvstring.cpp:5269:  [4] (format) vfprintf:
  If format strings can be influenced by an attacker, they can be exploited
  (CWE-134). Use a constant for the format specification.
lvstring.cpp:370:  [2] (buffer) memcpy:
  Does not check for buffer overflows when copying to destination (CWE-120).
  Make sure destination can always hold the source data.
lvstring.cpp:1072:  [2] (integer) atoi:
  Unless checked, the resulting number can exceed the expected range
  (CWE-190). If source untrusted, check both minimum and maximum, even if the
  input had no minus sign (large numbers can roll over into negative number;
  consider saving to an unsigned value if that is intended).
lvstring.cpp:1075:  [2] (integer) atoi:
  Unless checked, the resulting number can exceed the expected range
  (CWE-190). If source untrusted, check both minimum and maximum, even if the
  input had no minus sign (large numbers can roll over into negative number;
  consider saving to an unsigned value if that is intended).
lvstring.cpp:1126:  [2] (integer) atoi:
  Unless checked, the resulting number can exceed the expected range
  (CWE-190). If source untrusted, check both minimum and maximum, even if the
  input had no minus sign (large numbers can roll over into negative number;
  consider saving to an unsigned value if that is intended).
lvstring.cpp:1165:  [2] (integer) atoi:
  Unless checked, the resulting number can exceed the expected range
  (CWE-190). If source untrusted, check both minimum and maximum, even if the
  input had no minus sign (large numbers can roll over into negative number;
  consider saving to an unsigned value if that is intended).
lvstring.cpp:2517:  [2] (integer) atoi:
  Unless checked, the resulting number can exceed the expected range
  (CWE-190). If source untrusted, check both minimum and maximum, even if the
  input had no minus sign (large numbers can roll over into negative number;
  consider saving to an unsigned value if that is intended).
lvstring.cpp:3329:  [2] (buffer) MultiByteToWideChar:
  Requires maximum length in CHARACTERS, not bytes (CWE-120).
lvstring.cpp:3340:  [2] (buffer) MultiByteToWideChar:
  Requires maximum length in CHARACTERS, not bytes (CWE-120).
lvstring.cpp:3367:  [2] (buffer) char:
  Statically-sized arrays can be improperly restricted, leading to potential
  overflows or other issues (CWE-119!/CWE-120). Perform bounds checking, use
  functions that limit length, or ensure that the size is larger than the
  maximum possible length.
lvstring.cpp:3372:  [2] (buffer) char:
  Statically-sized arrays can be improperly restricted, leading to potential
  overflows or other issues (CWE-119!/CWE-120). Perform bounds checking, use
  functions that limit length, or ensure that the size is larger than the
  maximum possible length.
lvstring.cpp:3378:  [2] (buffer) char:
  Statically-sized arrays can be improperly restricted, leading to potential
  overflows or other issues (CWE-119!/CWE-120). Perform bounds checking, use
  functions that limit length, or ensure that the size is larger than the
  maximum possible length.
lvstring.cpp:5282:  [2] (misc) fopen:
  Check when opening files - can an attacker redirect it (via symlinks),
  force the opening of special file type (e.g., device files), move things
  around to create a race condition, control its ancestors, or change its
  contents? (CWE-362).
lvstring.cpp:5491:  [2] (buffer) memcpy:
  Does not check for buffer overflows when copying to destination (CWE-120).
  Make sure destination can always hold the source data.
lvstring.cpp:5548:  [2] (buffer) memcpy:
  Does not check for buffer overflows when copying to destination (CWE-120).
  Make sure destination can always hold the source data.
lvstring.cpp:5790:  [2] (integer) atoi:
  Unless checked, the resulting number can exceed the expected range
  (CWE-190). If source untrusted, check both minimum and maximum, even if the
  input had no minus sign (large numbers can roll over into negative number;
  consider saving to an unsigned value if that is intended).
lvstring.cpp:5792:  [2] (integer) atoi:
  Unless checked, the resulting number can exceed the expected range
  (CWE-190). If source untrusted, check both minimum and maximum, even if the
  input had no minus sign (large numbers can roll over into negative number;
  consider saving to an unsigned value if that is intended).
lvstring.cpp:5336:  [1] (buffer) strlen:
  Does not handle strings that are not \0-terminated; if given one it may
  perform an over-read (it could cause a crash if unprotected) (CWE-126).
lvstring.cpp:5368:  [1] (buffer) strlen:
  Does not handle strings that are not \0-terminated; if given one it may
  perform an over-read (it could cause a crash if unprotected) (CWE-126).

ANALYSIS SUMMARY:

Hits = 19
Lines analyzed = 5995 in approximately 0.10 seconds (59802 lines/second)
Physical Source Lines of Code (SLOC) = 5442
Hits@level = [0]   3 [1]   2 [2]  16 [3]   0 [4]   1 [5]   0
Hits@level+ = [0+]  22 [1+]  19 [2+]  17 [3+]   1 [4+]   1 [5+]   0
Hits/KSLOC@level+ = [0+] 4.04263 [1+] 3.49136 [2+] 3.12385 [3+] 0.183756 [4+] 0.183756 [5+]   0
Minimum risk level = 1

Not every hit is necessarily a security vulnerability.
You can inhibit a report by adding a comment in this form:
// flawfinder: ignore
Make *sure* it's a false positive!
You can use the option --neverignore to show these.

There may be other security vulnerabilities; review your code!
See 'Secure Programming HOWTO'
(https://dwheeler.com/secure-programs) for more information.
```

</details>

**Comment by benoit-pierre** [comment]
> As an aside, do you suppose it could be of interest to add a `flawfinder` check? https://dwheeler.com/flawfinder/
> Sample output on current master
> […]

I ran it locally, and it does not seem that useful: indiscriminately warning about every memcpy / strcpy call, every static array… 384 hits for crengine source files (including headers, without antiword & chmlib), that's a lot of false positives to wade through.

**Pull request merged**